### PR TITLE
Fixes #34767

### LINKED ISSUE #34767: Ensure that pkg.installed function refreshes properly with windows.
### Description of Issue/Question
A while back pkg.latest was fixed and I believe works properly but no one got around to fixing pkg.installed. In pkg.installed, if the version was 'latest', a refresh will happen no matter what
(possibly in addition to others) because of the lines

```
if salt.utils.is_true(kwargs.get('refresh', True)):
        refresh_db(saltenv)
```

in pkg.latest_version. Also if the version is not 'latest' and the cache was not already refreshed the state would fail because the call to pkg.list_pkgs (called by _find_install_targets) requires that the pkg database exist in order to work.

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

Makes sure that pkg.installed works in accordance with the docs when it comes to refreshing the pkg db in Windows. Even if, for some reason, it's not a perfect fix it will only improve things, and it shouldn't break anything non-Windows.
### What issues does this PR fix or reference?
saltstack/salt#34767
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
@hrumph Would you mind fixing this whitespace error, please? https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/3438/violations/file/salt/states/pkg.py/

**Comment by hrumph** [comment]
Last commit was needed because it's not only the call to pkg.installed that might cause a refresh so, the  test for refreshed had to be moved to the outer-most nesting level.

**Comment by cachedout** [comment]
@terminalmage Could you take a look here as well please?

File: salt/states/pkg.py [hunk 1]
```diff
@@ -1053,8 +1053,16 @@ def installed(
     if not isinstance(version, six.string_types) and version is not None:
         version = str(version)
 
+    refreshed = False
```
Comment (terminalmage) [inline salt/states/pkg.py]: Can we use something other than ``refreshed`` here? It's too close to ``refresh`` and will be confusing to maintain.

Comment (terminalmage) [inline salt/states/pkg.py]: Maybe ``already_refreshed``?

Comment (hrumph) [inline salt/states/pkg.py]: That sounds good to me. We'll go with that.

Comment (hrumph) [inline salt/states/pkg.py]: @terminalmage, would you be ok with "was_refreshed"? That way when we do the test we'll end up seeing
```
if os.path.isfile(rtag) and was_refreshed:
```
I think that "was_refreshed" captures the meaning a bit more. However, if you still insist on "already_refreshed" I'll go with that, as I said.

Comment (terminalmage) [inline salt/states/pkg.py]: Sure, that'd be fine. That makes it sufficiently different from "refresh", I think.

**Comment by hrumph** [comment]
Yo @cachedout this is ready to roll.

**Comment by hrumph** [comment]
Hi @cachedout. For good measure, I put the pkg.latest version in a try/except (same as is don in the "latest" function, and I also moved one line of code to a better stop (from the point of view of code elegance). Can you please merge this soon?

**Comment by hrumph** [comment]
@cachedout is there some reason why this isn't getting merged?

**Comment by cachedout** [comment]
Hi @hrumph. Sorry this fell off my radar. Thanks for the bump. I'm looking at it now.

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by hrumph** [comment]
@cachedout is there some reason why this can't be merged? I don't know what the failed tests mean.

**Comment by hrumph** [comment]
@cachedout why no merge?

**Comment by hrumph** [comment]
@cachedout can you give me some idea of what the typical merge schedule is? I would have thought that fixes for anything should get priority over all other pr's.

**Comment by hrumph** [comment]
@terminalmage can you help me to get this thing merged?

**Comment by terminalmage** [comment]
@hrumph Sorry, we've been quite busy working on the upcoming Carbon release, so this took longer to get merged than we would have hoped.

**Pull request merged**